test/EndToEndTests/Services/ODataDefaultService/DefaultWCFSvcDataSource.cs

```diff
@@ -1206,22 +1207,22 @@ protected override IEdmModel CreateModel()
             return DefaultInMemoryModel.CreateODataServiceModel("Microsoft.Test.OData.Services.ODataWCFService");
         }
 
-        protected override void ConfigureContainer(IContainerBuilder builder)
+        protected override void ConfigureContainer(IServiceCollection services)
         {
-            base.ConfigureContainer(builder);
-            builder.AddServicePrototype(new ODataMessageWriterSettings()
+            base.ConfigureContainer(services);
+            services.AddScoped(new ODataMessageWriterSettings()
             {
-                EnableWritingKeyAsSegment = false,
+                EnableReadingKeyAsSegment = false,
             });
 
-            builder.AddServicePrototype(new ODataMessageReaderSettings()
+            services.AddScoped(new ODataMessageWriterSettings()

```

Comment:
Was it intended to add `ODataMessageWriterSettings` twice? If so, please leave a comment in the code to explain why.

Reply:
We initially instantiated ODataMessageWriterSettings as a singleton, then generated clones of this instance as scoped, rather than consistently creating a new scope for ODataMessageWriterSettings. Upon evaluating the cloning process, it appeared to have negligible impact compared to generating a fresh instance of ODataMessageWriterSettings each time.